### PR TITLE
convert GetTags to Tags, update everywhere

### PR DESCRIPTION
**Pull request by davidbloss**
DRAFT: details to come...

## Changelog

- [ ] List your changes here
- [ ] Make a `changie` entry

### CODE REVIEW

File: actions_test.go [hunk 1]
```diff
@@ -447,12 +447,12 @@ func TestListExtendedTeamAccess(t *testing.T) {
...
 	requests := []TestRequest{
 		{
-			`{"query": "query ExtendedTeamAccessList($after:String!$first:Int!$input:IdentifierInput!){account{customActionsTriggerDefinition(input: $input){extendedTeamAccess(after: $after, first: $first){nodes{alias,id,aliases,contacts{address,displayName,id,type},group{alias,id},htmlUrl,manager{id,email,htmlUrl,name,role},members{nodes{id,email,htmlUrl,name,role},pageInfo{hasNextPage,hasPreviousPage,startCursor,endCursor},totalCount},name,responsibilities,tags{nodes{id,key,value},pageInfo{hasNextPage,hasPreviousPage,startCursor,endCursor},totalCount}},pageInfo{hasNextPage,hasPreviousPage,startCursor,endCursor},totalCount}}}}",
+			`{"query": "query ExtendedTeamAccessList($after:String!$first:Int!$input:IdentifierInput!){account{customActionsTriggerDefinition(input: $input){extendedTeamAccess(after: $after, first: $first){nodes{alias,id,aliases,contacts{address,displayName,id,type},group{alias,id},htmlUrl,manager{id,email,htmlUrl,name,role},members{nodes{id,email,htmlUrl,name,role},pageInfo{hasNextPage,hasPreviousPage,startCursor,endCursor},totalCount},name,responsibilities},pageInfo{hasNextPage,hasPreviousPage,startCursor,endCursor},totalCount}}}}",
```
Comment (rocktavious) [inline actions_test.go:450]: looks like you may have missed removing a `}` at the end

```suggestion
			`{"query": "query ExtendedTeamAccessList($after:String!$first:Int!$input:IdentifierInput!){account{customActionsTriggerDefinition(input: $input){extendedTeamAccess(after: $after, first: $first){nodes{alias,id,aliases,contacts{address,displayName,id,type},group{alias,id},htmlUrl,manager{id,email,htmlUrl,name,role},members{nodes{id,email,htmlUrl,name,role},pageInfo{hasNextPage,hasPreviousPage,startCursor,endCursor},totalCount},name,responsibilities,pageInfo{hasNextPage,hasPreviousPage,startCursor,endCursor},totalCount}}}}",
```

File: actions_test.go [hunk 2]
```diff
@@ -447,12 +447,12 @@ func TestListExtendedTeamAccess(t *testing.T) {
...
 		},
 		{
-			`{"query": "query ExtendedTeamAccessList($after:String!$first:Int!$input:IdentifierInput!){account{customActionsTriggerDefinition(input: $input){extendedTeamAccess(after: $after, first: $first){nodes{alias,id,aliases,contacts{address,displayName,id,type},group{alias,id},htmlUrl,manager{id,email,htmlUrl,name,role},members{nodes{id,email,htmlUrl,name,role},pageInfo{hasNextPage,hasPreviousPage,startCursor,endCursor},totalCount},name,responsibilities,tags{nodes{id,key,value},pageInfo{hasNextPage,hasPreviousPage,startCursor,endCursor},totalCount}},pageInfo{hasNextPage,hasPreviousPage,startCursor,endCursor},totalCount}}}}",
+			`{"query": "query ExtendedTeamAccessList($after:String!$first:Int!$input:IdentifierInput!){account{customActionsTriggerDefinition(input: $input){extendedTeamAccess(after: $after, first: $first){nodes{alias,id,aliases,contacts{address,displayName,id,type},group{alias,id},htmlUrl,manager{id,email,htmlUrl,name,role},members{nodes{id,email,htmlUrl,name,role},pageInfo{hasNextPage,hasPreviousPage,startCursor,endCursor},totalCount},name,responsibilities},pageInfo{hasNextPage,hasPreviousPage,startCursor,endCursor},totalCount}}}}",
```
Comment (rocktavious) [inline actions_test.go:455]: same here - miss removing a `}`

```suggestion
			`{"query": "query ExtendedTeamAccessList($after:String!$first:Int!$input:IdentifierInput!){account{customActionsTriggerDefinition(input: $input){extendedTeamAccess(after: $after, first: $first){nodes{alias,id,aliases,contacts{address,displayName,id,type},group{alias,id},htmlUrl,manager{id,email,htmlUrl,name,role},members{nodes{id,email,htmlUrl,name,role},pageInfo{hasNextPage,hasPreviousPage,startCursor,endCursor},totalCount},name,responsibilities,pageInfo{hasNextPage,hasPreviousPage,startCursor,endCursor},totalCount}}}}",
```

File: service.go [hunk 3]
```diff
@@ -97,7 +95,11 @@ func (s *Service) HasAlias(alias string) bool {
 }
 
 func (s *Service) HasTag(key string, value string) bool {
-	for _, tag := range s.Tags.Nodes {
+	tags, err := s.Tags(NewGQLClient(), nil)
```
Comment (rocktavious) [inline service.go:98]: 😬 we cannot do this.  This might not work depending on what kinds of configuration has happened.

This method should be moved to TagConnection

Comment (davidbloss) [reply service.go:98]: 🧐 Yes I see... this would _clearly_ result in chaos 😅

File: testdata/templates/custom_actions.tpl [hunk 4]
```diff
@@ -180,7 +180,6 @@
     "account": {
       "customActionsTriggerDefinition": {
         "extendedTeamAccess": {
-          "nodes": [ {{ template "team_1" }} ],
```
Comment (rocktavious) [inline testdata/templates/custom_actions.tpl:183]: why was this removed?  Don't we just need to remove "tags" not teams?

File: repository.go [hunk 5]
```diff
@@ -139,16 +137,8 @@ func (r *Repository) Hydrate(client *Client) error {
 		}
 	}
 
-	if r.Tags == nil {
-		r.Tags = &TagConnection{}
-	}
-	if r.Tags.PageInfo.HasNextPage {
```
Comment (rocktavious) [inline repository.go:145]: hmm this is going to present a problem for the CLI when doing `opslevel list repositories -o json`

Ugh man this is not a facet we considered 😮‍💨

File: tags.go [hunk 6]
```diff
@@ -230,34 +239,6 @@ func (client *Client) CreateTag(input TagCreateInput) (*Tag, error) {
 
 //#endregion
 
-//#region Retrieve
-
-// Deprecated: use client.GetServiceWithAlias(alias).Tags instead
```
Comment (rocktavious) [inline tags.go:235]: lets make sure we make changelog entries that we are removing these deprecated functions

File: testdata/templates/custom_actions.tpl (hunk 4 above)
Comment (davidbloss) [reply testdata/templates/custom_actions.tpl:183]: This was part of some crude troubleshooting and should be put back in
